### ISSUE TITLE
need recommendations for trusted data sources beyond Wikipedia for Knowledge submissions

### ISSUE THREAD

**Issue by lhawthorn** [instructlab/dev-docs#59]
For background on this issue, please review this document on [InstructLab knowledge submissions beyond Wikipedia](https://github.com/instructlab/dev-docs/blob/main/docs/knowledge-submissions-past-wikipedia.md). 

There are any number of other domains we may wish to add as approved content providers. Please use this issue to make recommendations for review by legal experts.

**Comment by lhawthorn** [comment]
It is not spectacularly helpful as a "domain" from which to draw from because lists must be filtered according to specific parameters and the outputs verified as not being in conflict with our [approved licenses list](https://github.com/instructlab/taxonomy/blob/main/CONTRIBUTING.md#legal) , but there are a number of open access journals from which we could and should draw cataloged on the [Directory of Open Access Journals website](https://doaj.org/).

82% of their submissions come from academic organizations, so it may behoove us to consider if we wish to scope our contributions in terms of domain (e.g. nasa.gov) versus publisher (e.g. Universidad de Chile).

It is also worth asking our legal team if sources that bear the [DOAJ Seal](https://doaj.org/apply/guide/#basic-criteria-for-inclusion)  meet our criteria for knowledge contributions provided they also fall under the scope of our acceptable licenses.

For example, [here are query results](https://doaj.org/search/journals?ref=homepage-box&source=%7B%22query%22%3A%7B%22bool%22%3A%7B%22must%22%3A%5B%7B%22terms%22%3A%7B%22index.license.exact%22%3A%5B%22CC%20BY%22%2C%22CC%20BY-SA%22%2C%22CC0%22%2C%22Public%20domain%22%5D%7D%7D%2C%7B%22query_string%22%3A%7B%22query%22%3A%22science%22%2C%22default_operator%22%3A%22AND%22%7D%7D%5D%7D%7D%2C%22track_total_hits%22%3Atrue%7D) where I have asked DOAJ to list out journals with the topic "science" where the content is licensed according to our approved licenses list rubric.

This list would need to be further vetted since some content is dual licensed in ways that would not work for inclusion per our approved licenses list (e.g. dual licensed CC BY and CC BY NC ND).

If we were to accept the DOAJ seal as part of our "approved content" rubric, it would rapidly open a number of open access publications for knowledge contributions.

**Comment by lhawthorn** [comment]
Would like to see InstructLab approve submissions from the [PLOS open access journals](https://plos.org/publish/) family of publications.

**Comment by lhawthorn** [comment]
Would like to see InstructLab approve submissions from the [OpenStax textbooks](https://openstax.org/subjects) family of publications.

**Comment by lhawthorn** [comment]
In an ideal universe, reviewing and adjudicating upon this list of proposed data sources and the list of data sources proposed already extant in the [Knowledge Submissions Past Wikipedia design document](https://github.com/instructlab/dev-docs/blob/main/docs/knowledge-submissions-past-wikipedia.md) would allow us to create and publish a rubric against which the triager team could automatically approve data sources for knowledge contributions to the project without requiring another round of review by legal experts. I believe such an outcome is possible and we should make it a project goal to get there. (This feedback may belong in it's own issue, but will file that one when the time is right.)

**Comment by jimmysjolund** [comment]
I would like to be able to add knowledge and skills based on the following (to start with):

- The Open Organization publications, https://theopenorganization.org/, all released under Creative Commons [Attribution-ShareAlike 4.0 International](https://creativecommons.org/licenses/by-sa/4.0/) license.
- Open Practice Library, https://openpracticelibrary.com/, licensed under a [Creative Commons Attribution 4.0](https://creativecommons.org/licenses/by-sa/4.0/) International license (could already have been requested by Donna Benjamin).
- Cynefin.io wiki, https://cynefin.io/wiki/Main_Page, "All material in this wiki is available under the [Creative Commons Attribution-ShareAlike License](https://creativecommons.org/licenses/by-sa/4.0/), unless otherwise indicated on a specific page."
- The Scrum Guide, https://scrumguides.org/index.html, license under the Attribution Share-Alike license of Creative Commons, accessible at https://creativecommons.org/licenses/by-sa/4.0/legalcode and also described in summary form at https://creativecommons.org/licenses/by-sa/4.0/. The ilab model do answer today that is has read the 2020 version of the Scrum Guide, so just as Wikipedia it has been pretrained, but could be improved.

**Comment by hpdempsey** [comment]
We would like to add knowledge from the Open Education Project.  https://research.redhat.com/blog/research_project/foundations-in-open-source-education/

**Comment by jimmysjolund** [comment]
> I would like to add knowledge from ibm.com.

Wouldn't that potentially break the ToS? https://www.ibm.com/legal/terms

**Comment by kwright15** [comment]
> > I would like to add knowledge from ibm.com.
> 
> Wouldn't that potentially break the ToS? https://www.ibm.com/legal/terms

I didn't see anything that prevents using the data in LLMs.

**Comment by jimmysjolund** [comment]
I'm not a lawyer but a couple of things stood out to me, such as:
"You may not mirror any of the content from this site on another Web site or in any other media. "

**Comment by kwright15** [comment]
We should check with Legal on that because it does say that web crawling is allowed as permitted in the robots.txt protocol, which seems to allow https://www.ibm.com/products/*. (My reason for wanting to use ibm.com is to add knowledge about IBM products.

**Comment by lhawthorn** [comment]
@kwright15 Thank you for your request. Following up from a conversation with Red Hat's legal team, we have been notified that inclusion of these materials would need to be accompanied by an approval statement from IBM's legal team. If that's important to you, we can help you make this ask and I would ask my fellow Community Maintainers @joesepi @jjasghar and @mmcelaney for their assistance.

**Comment by lhawthorn** [comment]
Folks, following meetings with knowledgeable legal folks, the following sources are now approved for Knowledge submissions:

- Project Gutenberg: https://www.gutenberg.org/	
- Wikisource: https://en.wikisource.org/
- OpenStax textbooks family of publications: https://openstax.org/subjects 
- The Open Organization publications: https://theopenorganization.org/

We further have the following guidance:
While it seems easy to say that a site on e.g. a .gov domain means the content falls into the public domain, it is actually not that simple as different subsections or, for example, images, may fall under different licensing schemes.

Further, there are a number of times where we find that the metadata for a particular document may be incorrect and therefore a blanket approval for content from a site may yield incorrect assumptions about whether or not we should take in this data as a knowledge source, e.g. Internet Archive searches often return results listed as out of copyright, e.g. pre-1927 works,  but that metadata is inaccurate.

Lastly, there are a number of sites detailed in this requested list that appear to only house openly licensed data sources that appear in our [list of acceptable content licenses](https://github.com/instructlab/community/blob/main/FAQ.md#my-contribution-requires-submitting-data-along-with-code-what-data-is-permissible-to-include), but *some* of the content linked therein is not actually under an open content license. Because of this detail, it is not possible to provide a blanket approval to use content at that site.

Bottom line in the discussion was that even more knowledge submissions from various data sources will require hand vetting of each incoming data source to ensure it meets our acceptable license criteria. This does not necessarily require a legal professional to do, but does put the onus upon the Triaging team and/or other maintainers of the Taxonomy repo to understand the licensing criteria and to vet that incoming content.

I have already volunteered as tribute for the Triage team to help them with these content licensing questions, mostly because I am one of those strange people who actually finds this sort of work interesting. :D

**Comment by lhawthorn** [comment]
I hope to have made @jimmysjolund had a happier day with some of the updates in this issue and the related PR to update our dev docs. I am going to close this issue when #105 is reviewed and merged because we will take in further requests for reviews of different content via a Pull Request against the dev doc updated in #105.

**Comment by kwright15** [comment]
> @kwright15 Thank you for your request. Following up from a conversation with Red Hat's legal team, we have been notified that inclusion of these materials would need to be accompanied by an approval statement from IBM's legal team. If that's important to you, we can help you make this ask and I would ask my fellow Community Maintainers @joesepi @jjasghar and @mmcelaney for their assistance.

Thank you @lhawthorn! I am interested in your help to get approval from IBM's legal team to contribute ibm.com knowledge. I am happy to reach out to the legal team, I'm just now sure who to start with.

**Comment by lhawthorn** [comment]
@kwright15 I am sorry I missed your note until today, please reach out to me via email for help or ping me in the InstructLab Slack workspace.
